File: InterviewPreparationKit/QueueUsingStacks.py
# https://www.hackerrank.com/challenges/ctci-queue-using-two-stacks/problem
class MyQueue(object):
    def __init__(self):
        self.stack_newest_on_top = []
        self.stack_oldest_on_top = []
    
    def peek(self):
        if len(self.stack_oldest_on_top) == 0:
            while len(self.stack_newest_on_top) > 0:
                self.stack_oldest_on_top.append(self.stack_newest_on_top.pop())
        return self.stack_oldest_on_top[-1]
        
    def pop(self):
        if len(self.stack_oldest_on_top) == 0:
            while len(self.stack_newest_on_top) > 0:
                self.stack_oldest_on_top.append(self.stack_newest_on_top.pop())
        return self.stack_oldest_on_top.pop()
        
    def put(self, value):
        self.stack_newest_on_top.append(value)


queue = MyQueue()
t = int(input())
for line in range(t):
    values = map(int, input().split())
    values = list(values)
    if values[0] == 1:
        queue.put(values[1])        
    elif values[0] == 2:
        queue.pop()
    else:
        print(queue.peek())

File: InterviewPreparationKit/QueueUsingStacks.js
// https://www.hackerrank.com/challenges/ctci-queue-using-two-stacks/problem
const MyQueue = class {
    constructor() {
        this.stackNewestOnTop = []
        this.stackOldestOnTop = []
    }
    
    enqueue(value) {
        this.stackNewestOnTop.push(value);
    }
    
    peek() {
        if (this.stackOldestOnTop.length == 0) {
            while (this.stackNewestOnTop.length > 0) {
                this.stackOldestOnTop.push(this.stackNewestOnTop.pop());
            }
        }
        
        return this.stackOldestOnTop[this.stackOldestOnTop.length - 1];
    }
    
    dequeue() {
        if (this.stackOldestOnTop.length == 0) {
            while (this.stackNewestOnTop.length > 0) {
                this.stackOldestOnTop.push(this.stackNewestOnTop.pop());
            }
        }
        
        return this.stackOldestOnTop.pop();
    }
};


function processData(input) {
    let queue = new MyQueue();
    let n = parseInt(input[0]);
    
    for (let i = 1; i <= n; i++) {
        let array = input[i].split(' ');
        let operation = array[0];
        if (operation == 1) {
            queue.enqueue(array[1]);
        }
        else if (operation == 2) {
            queue.dequeue();
        }
        else if (operation == 3) {
            console.log(queue.peek());
        }
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    _input = _input.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));
        
    processData(_input);
});
